state/mapstate/map_state.go

```diff
@@ -15,7 +15,7 @@ import (
 
 // Version is the map state Version. States with old versions should
 // perform an upgrade before.
-const Version = 2
+const Version = 3
```

Comment:
I think if you rebase on master, this probably causes an error message when launching cluster and ask the user to upgrade. You'd need to add a migration here:

https://github.com/ipfs/ipfs-cluster/blob/master/state/mapstate/migrate.go#L17

What the migration is going to do is simply pretty much the same as Unmarshal (https://github.com/ipfs/ipfs-cluster/blob/master/state/mapstate/map_state.go#L159). But it needs to do it there in the migrate code (and it cannot re-use unmarshal because if it changed in the future, the migration from version 2 of the state would fail).

We just added migration handling, so I just think this is an opportunity to give it a test. Would you be up for it? Otherwise, just leave Version = 2, which will work anyways (I think).

Reply:
Mostly in agreement.  I think this may cause issues when starting with existing v2 formatted raft state even if you keep `Version = 2`.  I had some issues getting the serialized v1 state to decode into a v2 state, and I suspect the change to pinSerial here will cause a similar problem between v2 and v2'.  But as the current v2 format has been out for under a day and not bundled into a release, I'm doubting such a break is a real problem.

Reply:
Yes, I'll look into upping to `Version = 3` and adding a migration.

Reply:
So I've started a clean single-node cluster on master and pinned an item. I then shutdown the cluster, switched to feat/add-pin-name with `Version = 2` and did not observe an issue on startup or listing pins.

Doing the same (without a migration) except keeping the `Version = 3` change appears to cause an issue where I am unable to list items I previously had pinned.

I'll see what happens in the second scenario after I add a migration, but would it be best to leave `Version = 2` if that doesn't cause issues?